src/_P092_DLbus.ino

```diff
@@ -0,0 +1,1180 @@
+          CurIdx = getFormItemInt(plugin_092_Indizes[i]);
+          if (CurIdx < 1) {
+            CurIdx = 1;
+          }
+          PCONFIG(i + 1) = (OptionIdx << 8) + CurIdx;
+          ExtraTaskSettings.TaskDeviceValueDecimals[event->BaseVarIndex + i] = P092_OptionValueDecimals[OptionIdx];
+        }
+
+#ifdef PLUGIN_092_DEBUG
+        if (loglevelActiveFor(LOG_LEVEL_INFO)) {
+	        P092_DLB_Pin = CONFIG_PIN1;
+
+	        String log = F("PLUGIN_WEBFORM_SAVE :");
+	        log += F(" DLB_Pin:");
+	        log += P092_DLB_Pin;
+	        log += F(" MinPulseWidth:");
+	        log += DataSettings.MinPulseWidth;
+	        log += F(" MaxPulseWidth:");
+	        log += DataSettings.MaxPulseWidth;
+	        log += F(" MinDoublePulseWidth:");
+	        log += DataSettings.MinDoublePulseWidth;
+	        log += F(" MaxDoublePulseWidth:");
+	        log += DataSettings.MaxDoublePulseWidth;
+	        log += F(" IdxSensor:");
+	        log += DataSettings.IdxSensor;
+	        log += F(" IdxExtSensor:");
+	        log += DataSettings.IdxExtSensor;
+	        log += F(" IdxOutput:");
+	        log += DataSettings.IdxOutput;
+	        if (DataSettings.SpeedBytes > 0) {
+	          log += F(" IdxDrehzahl:");
+	          log += DataSettings.IdxDrehzahl;
+	        }
+	        if (DataSettings.AnalogBytes > 0) {
+	          log += F(" IdxAnalog:");
+	          log += DataSettings.IdxAnalog;
+	        }
+	        if (DataSettings.MaxHeatMeters > 0) {
+	          log += F(" IdxHmRegister:");
+	          log += DataSettings.IdxHmRegister;
+	        }
+	        if (DataSettings.VolumeBytes > 0) {
+	          log += F(" IdxVolume:");
+	          log += DataSettings.IdxVolume;
+	        }
+	        if (DataSettings.MaxHeatMeters > 0) {
+	          log += F(" IdxHM1:");
+	          log += DataSettings.IdxHeatMeter1;
+	          log += F(" IdxkWh1:");
+	          log += DataSettings.IdxkWh1;
+	          log += F(" IdxMWh1:");
+	          log += DataSettings.IdxMWh1;
+	        }
+	        if (DataSettings.MaxHeatMeters > 1) {
+	          log += F(" IdxHM2:");
+	          log += DataSettings.IdxHeatMeter2;
+	          log += F(" IdxkWh2:");
+	          log += DataSettings.IdxkWh2;
+	          log += F(" IdxMWh2:");
+	          log += DataSettings.IdxMWh2;
+	        }
+	        if (DataSettings.MaxHeatMeters > 2) {
+	          log += F(" IdxHM3:");
+	          log += DataSettings.IdxHeatMeter3;
+	          log += F(" IdxkWh3:");
+	          log += DataSettings.IdxkWh3;
+	          log += F(" IdxMWh3:");
+	          log += DataSettings.IdxMWh3;
+	        }
+	        log += F(" IdxCRC:");
+	        log += DataSettings.IdxCRC;
+	        addLog(LOG_LEVEL_INFO, log);
+				}
+#endif  // PLUGIN_092_DEBUG
+        UserVar[event->BaseVarIndex] = NAN;
+        success = true;
+        break;
+      }
+
+    case PLUGIN_INIT:
+      {
+        if (P092_init) {
+          addLog(LOG_LEVEL_ERROR, F("P092_init -> Already done!"));
+          return true;
```

Comment:
what if someone change pin numer in configuration?

Reply:
What do you mean?
P092_DLB_Pin is CONFIG_PIN1.
UserVar[event->BaseVarIndex] = NAN is used to show a NAN in the web device page.

Reply:
but if you return here and pin number is changed, plugin will not attachInterrupt to new pin

Reply:
Yep, you have to restart the node to apply new settings with code like this.
The init is not cleared and you will enter this init call when you save the settings of the task.

Reply:
Done.
A new init will be started if the device pin has changed.